src/utils/relativeId.ts

```diff
@@ -11,7 +11,7 @@ export default function relativeId(id: string) {
 	return relative(resolve(), id);
 }
 
-export function isPlainPathFragment(name: string) {
+export function isPlainPathFragment(name: string, sanitizeFileName = defaultSanitizeFileName) {
```

Comment:
Here, the option should also not be optional.

Reply:
Looking further through the code, I see that the problem is that this function can also be called during the build phase when emitting assets. My feeling is that we should just remove the sanitation check here as the final file names will be sanitised again anyway.

Reply:
The problem is that the default version is used for validation in two separate places in `FileEmitter.ts`.

I would personally prefer to remove these validations entirely and allow any user defined schemes, but didn't want to push that here necessarily.

Passing the sanitization function exactly through all these validation routes seems to be the alternative, let me know which you prefer.

Reply:
Yes, in `FileEmitter` you would basically need to extend the `OutputSpecificFileData` type. But I think it would be worth it to do that so that it is consistent everywhere.

Reply:
Ok I've added this wiring.